router/internal/eventorder/eventorder.go

```diff
@@ -0,0 +1,255 @@
+package eventorder
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"sync"
+)
+
+type State string
+
+const (
+	SucceededState State = "succeeded"
```

Comment:
Probably we can use the jobsdb states here rather than re initialising them again

Reply:
The initial intention was to keep it clean and not introduce any other dependency apart from the standard libraries. Keeping it safe from import cycles too.

Reply:
Probably then move it from jobs db too to another common types folder. We might add a few more states and might miss to update it here